homeassistant/components/legrandinone.py

```diff
@@ -0,0 +1,480 @@
+            data['type'] = 'bus_command'
+
+        data['values'] = call.data.get(CONF_DIMENSION_VALUES)
+
+        if not await LegrandInOneCommand.send_command(data):
+            _LOGGER.error('Failed LegrandInOne command for %s', str(call.data))
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_SEND_COMMAND, async_send_command,
+        schema=SEND_COMMAND_SCHEMA)
+
+    @callback
+    def event_callback(event):
+        """Handle incoming legrandinone events.
+
+        IOBL events arrive as dictionaries of varying content
+        depending on their type. Identify the events and distribute
+        accordingly.
+        """
+        event_type = event.get('type')
+        _LOGGER.debug('event of type %s: %s', event_type, event)
+
+        # Don't propagate non entity events (eg: version string, ack response)
+        if event_type not in hass.data[DATA_ENTITY_LOOKUP]:
+            _LOGGER.debug('unhandled event of type: %s', event_type)
+            return
+
+        # Lookup entities who registered this device id as device id or alias
+        event_id = event.get('legrand_id', None)
+
+        entity_ids = hass.data[DATA_ENTITY_LOOKUP][event_type][event_id]
+
+        if entity_ids:
+            # Propagate event to every entity matching the device id
+            for entity in entity_ids:
+                _LOGGER.debug('passing event to %s', entity)
+                async_dispatcher_send(hass,
+                                      SIGNAL_HANDLE_EVENT.format(entity),
+                                      event)
+        else:
+            device_type = event.get('who')
+
+            # If device is not yet known, register with platform (if loaded)
+            if event_type in hass.data[DATA_DEVICE_REGISTER]:
+                if device_type in hass.data[DATA_DEVICE_REGISTER][event_type]:
+                    _LOGGER.debug('device_id not known, adding new device')
+
+                    hass.data[DATA_ENTITY_LOOKUP][event_type][
+                        event_id].append(TMP_ENTITY.format(event_id))
+                    hass.async_create_task(
+                        hass.data[DATA_DEVICE_REGISTER][event_type][
+                            device_type](event))
+
+                else:
+                    _LOGGER.debug(
+                        'device_id not known and automatic %s add disabled',
+                        device_type)
+            else:
+                _LOGGER.debug('device_id not known and automatic add disabled')
+
+    # When connecting to tcp host instead of serial port (optional)
+    host = config[DOMAIN].get(CONF_HOST)
+    # TCP port when host configured, otherwise serial port
+    port = config[DOMAIN][CONF_PORT]
+
+    @callback
+    def reconnect(exc=None):
+        """Schedule reconnect after connection has been unexpectedly lost."""
+        # Reset protocol binding before starting reconnect
+        hass.data[IOBL_PROTOCOL_HANDLE] = None
```

Comment:
We don't need to do this. Pass the new protocol with the signal availability dispatch call instead.

Reply:
Done.